### PR TITLE
Prune files within symlinked paths

### PR DESCRIPTION
**Pull request by dblandin**
Despite documentation from `node-glob` which states that symlinked
directories are not followed and crawled, that doesn't seem to be the
case, causing errors when listing files on CC or validating issues
emitted from ESLint regarding ignored files. This adds behavior to
manually prune any path within a symlink before analysis begins.

### CODE REVIEW

File: bin/eslint.js [hunk 1]
```diff
@@ -56,18 +56,17 @@ function contentBody(check) {
...
     stats.isFile() &&
-    !stats.isSymbolicLink()
-    && extensions.indexOf(extension) >= 0
+    !stats.isSymbolicLink() &&
+    extensions.indexOf(extension) >= 0
   );
 }
 
-function isFileIgnoredByLibrary(file) {
-  var path = file.replace(/^\/code\//, "");
```
Comment (GordonDiggs) [inline bin/eslint.js]: I found without this `replace`, the call to `isPathIgnored` did not return correctly. Can you confirm that this still works?

Comment (dblandin) [inline bin/eslint.js]: @GordonDiggs Just simplified this PR without the drive-by refactoring.

Comment (dblandin) [inline bin/eslint.js]: One of the file strategies has a hard-coded `/code/` prefix which likely relies on that replace for the ignore logic to behave as intended.

File: bin/eslint.js [hunk 2]
```diff
@@ -76,6 +76,18 @@ function isFileIgnoredByLibrary(file) {
   return ignored;
 }
 
+function prunePathsWithinSymlinks(paths) {
```
Comment (GordonDiggs) [inline bin/eslint.js]: Can you explain what this function is doing?

Comment (dblandin) [inline bin/eslint.js]: Sure, this is extracting all of the symlinks from `paths` and filtering out any paths which contain the symlink at the front of the path.

For example, if we have
```
a/       -> not a symlink
a/a.txt  -> not a symlink
b/       -> symlink
b/b.txt  -> not a symlink
```

we'll be left with
```js
["a/", "a/a.txt"]
```

Comment (GordonDiggs) [inline bin/eslint.js]: Cool - might be good to add a comment to that effect. WDYT?

**Comment by GordonDiggs** [comment]
LGTM
